### PR TITLE
Energy monitor V0.1.0

### PR DESCRIPTION
**Pull request by JiacongSun**
## Add energy monitor RTL V0.1.0 and its testbench in a separate folder.
**Function:**
- A synchronous module computes the Hamiltonian energy.
- An enable signal _en_i_ controls if the module logics and built-in registers work or not.

**Performance:**
- Latency: (_DATASPIN/PARALLELISM+PIPESMID+1_) clock cycles.

**Configurable parameters:**
- Data width of J, H, bias scaling factor, and number of spins (_DATASPIN_)
- Parallelism (_PARALLELISM_) of MAC units (affects the input width of J/H/scaling factor)
- Pipeline depth at the module input interface (_PIPESINTF_).
- Pipeline depth within the adder tree output interface (_PIPESMID_).

**Verification:**
- Simple cases, corner cases, and random cases are all verified. All test modes are integrated into the testbench.
- An autotest script in Python is provided in ./utils/autotest_energy_monitor.py.
- The details of verified cases are listed in the [README](https://github.com/KULeuven-MICAS/lagd-im/tree/energy_monitor/hw/unit_tests/energy_monitor) of the testbench folder.

### CODE REVIEW

File: hw/rtl/energy_monitor/logic_ctrl.sv [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    input logic cmpt_done_i,
+
+    output logic energy_valid_o,
+    input logic energy_ready_i,
+
+    input logic debug_en_i
+);
+    // State enumeration
+    typedef enum logic [1:0] {
+        SLEEP = 2'b00,
```
Comment (clmcsn) [inline hw/rtl/energy_monitor/logic_ctrl.sv]: What is the difference between IDLE and SLEEP? As we don't do DVFS, they might be the same?
